modules/gdscript/gdscript.cpp

```diff
@@ -603,10 +603,8 @@ Error GDScript::reload(bool p_keep_state) {
 		}
 		if (!source_path.empty()) {
 			MutexLock lock(GDScriptCache::singleton->lock);
-			Ref<GDScript> self(this);
 			if (!GDScriptCache::singleton->shallow_gdscript_cache.has(source_path)) {
-				GDScriptCache::singleton->shallow_gdscript_cache[source_path] = self;
-				self->unreference();
```

Comment:
I think that `self.unref()` was meant to be used in this case, but then again it's not really needed to do this manually:

https://github.com/godotengine/godot/blob/0ee0fa42e6639b6fa474b7cf6afc6b1a78142185/core/reference.h#L233-L235

Also found some interesting remark for mutexes:

https://github.com/godotengine/godot/blob/0ee0fa42e6639b6fa474b7cf6afc6b1a78142185/core/reference.h#L216-L225